sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala

```diff
@@ -936,6 +936,35 @@ class SQLQuerySuite extends QueryTest with SQLTestUtils with TestHiveSingleton {
     assert(e.getMessage.contains("Distinct window functions are not supported"))
   }
 
+  test("window function: better support of parentheses") {
+    val data = Seq(
+      WindowData(1, "a", 5),
+      WindowData(2, "a", 6),
+      WindowData(3, "b", 7),
+      WindowData(4, "b", 8),
+      WindowData(5, "c", 9),
+      WindowData(6, "c", 10)
+    )
+    sparkContext.parallelize(data).toDF().registerTempTable("windowData")
+
+    checkAnswer(
+      sql(
+        """
+          |select month, area, product,
+          |sum(product + 1) over (partition by ((1) + (1 - 1) -
+          |(2 * 1 / 2) + (1) + product - (product)) order by 2)
```

Comment:
Putting this query into the test is because we want to make sure some corner cases passed. E.g., (expression) op (expression op expression). I will try simpler ones.